File: problems/557-Reverse-Words-in-a-String-III/solution.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        for (int i = 0; i < words.length; i++) {
            char[] chars = words[i].toCharArray();
            for (int j = 0; j < chars.length / 2; j++) {
                chars[j] ^= chars[chars.length-1-j];
                chars[chars.length-1-j] ^= chars[j];
                chars[j] ^= chars[chars.length-1-j];
            }
            words[i] = new String(chars);
        }

        String result = String.join(" ", words);
        return result;
    }
}

File: problems/557-Reverse-Words-in-a-String-III/solution.go
func reverse(s string) string {
    result := ""
    for i := len(s)-1; i >= 0; i-- {
        result += string(s[i])
    }

    return result
}

func reverseWords(s string) string {
    words := strings.Split(s, " ")

    result := ""
    for i := 0; i < len(words)-1; i++ {
        result += reverse(words[i]) + " "
    }
    result += reverse(words[len(words)-1])

    return result
}

